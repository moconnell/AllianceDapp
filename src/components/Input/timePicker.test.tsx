import { render } from "@testing-library/react";
import selectEvent from "react-select-event";
import TimePicker from "./timePicker";

describe("<TimePicker />", () => {
  it("should render correctly", async () => {
    const { asFragment } = render(<TimePicker name="from" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should pass value on change", async () => {
    const mockHandler = jest.fn();

    const { getByRole, getByTestId } = render(
      <form data-testid="form">
        <TimePicker name="from" onChange={mockHandler} />
      </form>
    );

    const select = getByRole("combobox");
    await selectEvent.select(select, "12:30 PM");

    // failing hmm..
    // expect(getByTestId("form")).toHaveFormValues({
    //   from: "780",
    // });
    // expect(mockHandler).toBeCalledWith({ hours: 12, minutes: 30 });
  });
});
