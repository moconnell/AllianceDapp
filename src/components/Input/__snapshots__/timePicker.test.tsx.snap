// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimePicker /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-select__wrapper css-42b2qy"
  >
    <select
      class="chakra-select css-1ik61og"
      name="from"
    >
      <option
        value="0"
      >
        12:00 AM
      </option>
      <option
        value="30"
      >
        12:30 AM
      </option>
      <option
        value="60"
      >
        1:00 AM
      </option>
      <option
        value="90"
      >
        1:30 AM
      </option>
      <option
        value="120"
      >
        2:00 AM
      </option>
      <option
        value="150"
      >
        2:30 AM
      </option>
      <option
        value="180"
      >
        3:00 AM
      </option>
      <option
        value="210"
      >
        3:30 AM
      </option>
      <option
        value="240"
      >
        4:00 AM
      </option>
      <option
        value="270"
      >
        4:30 AM
      </option>
      <option
        value="300"
      >
        5:00 AM
      </option>
      <option
        value="330"
      >
        5:30 AM
      </option>
      <option
        value="360"
      >
        6:00 AM
      </option>
      <option
        value="390"
      >
        6:30 AM
      </option>
      <option
        value="420"
      >
        7:00 AM
      </option>
      <option
        value="450"
      >
        7:30 AM
      </option>
      <option
        value="480"
      >
        8:00 AM
      </option>
      <option
        value="510"
      >
        8:30 AM
      </option>
      <option
        value="540"
      >
        9:00 AM
      </option>
      <option
        value="570"
      >
        9:30 AM
      </option>
      <option
        value="600"
      >
        10:00 AM
      </option>
      <option
        value="630"
      >
        10:30 AM
      </option>
      <option
        value="660"
      >
        11:00 AM
      </option>
      <option
        value="690"
      >
        11:30 AM
      </option>
      <option
        value="720"
      >
        12:00 PM
      </option>
      <option
        value="750"
      >
        12:30 PM
      </option>
      <option
        value="780"
      >
        1:00 PM
      </option>
      <option
        value="810"
      >
        1:30 PM
      </option>
      <option
        value="840"
      >
        2:00 PM
      </option>
      <option
        value="870"
      >
        2:30 PM
      </option>
      <option
        value="900"
      >
        3:00 PM
      </option>
      <option
        value="930"
      >
        3:30 PM
      </option>
      <option
        value="960"
      >
        4:00 PM
      </option>
      <option
        value="990"
      >
        4:30 PM
      </option>
      <option
        value="1020"
      >
        5:00 PM
      </option>
      <option
        value="1050"
      >
        5:30 PM
      </option>
      <option
        value="1080"
      >
        6:00 PM
      </option>
      <option
        value="1110"
      >
        6:30 PM
      </option>
      <option
        value="1140"
      >
        7:00 PM
      </option>
      <option
        value="1170"
      >
        7:30 PM
      </option>
      <option
        value="1200"
      >
        8:00 PM
      </option>
      <option
        value="1230"
      >
        8:30 PM
      </option>
      <option
        value="1260"
      >
        9:00 PM
      </option>
      <option
        value="1290"
      >
        9:30 PM
      </option>
      <option
        value="1320"
      >
        10:00 PM
      </option>
      <option
        value="1350"
      >
        10:30 PM
      </option>
      <option
        value="1380"
      >
        11:00 PM
      </option>
      <option
        value="1410"
      >
        11:30 PM
      </option>
    </select>
    <div
      class="chakra-select__icon-wrapper css-162mkon"
    >
      <svg
        aria-hidden="true"
        class="chakra-select__icon"
        focusable="false"
        role="presentation"
        style="width: 1em; height: 1em; color: currentColor;"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
